cmd/crates/soroban-rpc/src/txn.rs

```diff
@@ -315,13 +315,17 @@ pub fn assemble(
 
     // update the fees of the actual transaction to meet the minimum resource fees.
     let classic_transaction_fees = crate::DEFAULT_TRANSACTION_FEES;
-    // Pad the fees up by 15% for a bit of wiggle room.
-    tx.fee = (tx.fee.max(
-        classic_transaction_fees
-            + u32::try_from(simulation.min_resource_fee)
-                .map_err(|_| Error::LargeFee(simulation.min_resource_fee))?,
-    ) * 115)
+
+    // Pad the fees up by 15% for a bit of wiggle room
+    let padded_fee = u64::from(
+        tx.fee.max(
+            classic_transaction_fees
+                + u32::try_from(simulation.min_resource_fee)
+                    .map_err(|_| Error::LargeFee(simulation.min_resource_fee))?,
+        ),
+    ) * 115
         / 100;
+    tx.fee = u32::try_from(padded_fee).map_err(|_| Error::LargeFee(padded_fee))?;
```

Comment:
I don't think it's correct to return an error if the padded result overflows; if the simulated fee fits into `u32`, then I don't think we should return an error. I.e. I think we should clamp the max fee at `u32::MAX` (as long as simulated fee fits it, that is).

Reply:
yeah fair enough